monai/apps/deepgrow/dataset.py

```diff
@@ -0,0 +1,271 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import os
+import sys
+from typing import Callable, Dict, List, Sequence, Union
+
+import numpy as np
+
+from monai.apps.datasets import DecathlonDataset
+from monai.transforms import AsChannelFirstd, Compose, LoadImaged, Orientationd, Spacingd
+from monai.utils import GridSampleMode
+
+
+# TODO:: Test basic functionality
+# TODO:: Unit Test
+class DeepgrowDataset(DecathlonDataset):
+    def __init__(
+        self,
+        dimension: int,
+        pixdim: Sequence[float],
+        root_dir: str,
+        task: str,
+        section: str,
+        transform: Union[Sequence[Callable], Callable] = (),
+        download: bool = False,
+        seed: int = 0,
+        val_frac: float = 0.2,
+        cache_num: int = sys.maxsize,
+        cache_rate: float = 1.0,
+        num_workers: int = 0,
+        limit: int = 0,
+    ) -> None:
```

Comment:
maybe we want a docstring here to explain these arguments?
and I think it might be good to check that dimension should be either 2 or 3

I don't understand the limit args here

Reply:
fixed.. added the docs